### PR TITLE
feat(rpc): implement Filecoin.NetFindPeer

### LINKED ISSUE #4562: Filecoin.NetFindPeer

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- implement `Filecoin.NetFindPeer`
- fix `Filecoin.NetPeers` result to match lotus (multiaddr without p2p part)
- move `net` types to a separate file

Manual test result

forest
```json
{"jsonrpc":"2.0","id":0,"result":{"ID":"12D3KooWABQ5gTDHPWyvhJM7jPhtNwNJruzTEo32Lo4gcS5ABAMm","Addrs":["/dns/bootstrap-calibnet-0.chainsafe-fil.io/tcp/34000"]}}
```
lotus
```json
{"id":0,"jsonrpc":"2.0","result":{"ID":"12D3KooWABQ5gTDHPWyvhJM7jPhtNwNJruzTEo32Lo4gcS5ABAMm","Addrs":["/ip4/148.113.17.89/tcp/34000","/ip6/2402:1f00:8300:959::/tcp/34000"]}}
```

## Reference issue to close (if applicable)

Closes #4562

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by aatifsyed — APPROVED** [review]

File: src/libp2p/service.rs [hunk 1]
```diff
@@ -484,6 +485,12 @@ async fn handle_network_message(
                         warn!("Failed to get Libp2p listeners");
                     }
                 }
+                NetRPCMethods::Peer(response_channel, peer) => {
+                    let addresses = swarm.behaviour().peer_addresses().get(&peer).cloned();
+                    if response_channel.send(addresses).is_err() {
+                        warn!(%peer, "Failed to get Libp2p peer addresses");
```
Comment (LesnyRumcajs) [inline src/libp2p/service.rs:491]: Is it a failure in obtaining the p2p address or is this a channel fiasco, e.g., dropped?

File: scripts/tests/api_compare/filter-list-offline [hunk 2]
```diff
@@ -8,6 +8,7 @@
 !Filecoin.NetAgentVersion
 !Filecoin.NetAutoNatStatus
 !Filecoin.NetPeers
+!Filecoin.NetFindPeer
```
Comment (LesnyRumcajs) [inline scripts/tests/api_compare/filter-list-offline:11]: Where is this method invoked in the tests?

Comment (hanabi1224) [reply scripts/tests/api_compare/filter-list-offline:11]: Not invoked at this moment but it's not available on an offline server

File: src/libp2p/service.rs (hunk 1 above)
Comment (hanabi1224) [reply src/libp2p/service.rs:491]: It's a channel fiasco. The oneshot API does not return an error tho
```rust
pub fn send(self, t: T) -> Result<(), T>
```

File: scripts/tests/api_compare/filter-list-offline (hunk 2 above)
Comment (LesnyRumcajs) [reply scripts/tests/api_compare/filter-list-offline:11]: Why is it not invoked?

File: src/libp2p/service.rs (hunk 1 above)
Comment (LesnyRumcajs) [reply src/libp2p/service.rs:491]: the error message should be adjusted, then, correct?

File: scripts/tests/api_compare/filter-list-offline (hunk 2 above)
Comment (hanabi1224) [reply scripts/tests/api_compare/filter-list-offline:11]: With the current testing infra, it's hard to automate the testing for this one

Comment (LesnyRumcajs) [reply scripts/tests/api_compare/filter-list-offline:11]: Why?

File: src/libp2p/service.rs (hunk 1 above)
Comment (hanabi1224) [reply src/libp2p/service.rs:491]: Fixed for all the variants by switching to `flume`

File: scripts/tests/api_compare/filter-list-offline (hunk 2 above)
Comment (LesnyRumcajs) [reply scripts/tests/api_compare/filter-list-offline:11]: Can we do at least "unhappy" path test? That is, to find a peer that we _know_ doesn't exist?

**Review by LesnyRumcajs — APPROVED** [review]

File: scripts/tests/api_compare/filter-list-offline (hunk 2 above)
Comment (hanabi1224) [reply scripts/tests/api_compare/filter-list-offline:11]: `.ignore("It times out in lotus when peer not found")`

**Pull request merged**